middleware/src/middleware.go

```diff
@@ -81,27 +88,48 @@ func (middleware *Middleware) demoCLightningRPC() {
 	nodeinfo, err := ln.Call("getinfo")
 	if err != nil {
 		log.Println(err.Error() + " Lightningd getinfo called failed.")
-	} else {
-		middleware.info.LightningAlias = nodeinfo.Get("alias").String()
+		return
 	}
+	middleware.info.LightningAlias = nodeinfo.Get("alias").String()
 }
 
 func (middleware *Middleware) rpcLoop() {
 	for {
 		middleware.demoBitcoinRPC()
 		middleware.demoCLightningRPC()
-		middleware.events <- &middleware.info
+		outgoing := &basemessages.Outgoing{
+			Outgoing: &basemessages.Outgoing_MiddlewareInfo{
+				MiddlewareInfo: &basemessages.MiddlewareInfoOutgoing{
+					Blocks:         middleware.info.Blocks,
+					Difficulty:     float32(middleware.info.Difficulty),
+					LightningAlias: middleware.info.LightningAlias,
+				},
+			},
+		}
+		middleware.mu.Lock()
+		middleware.events <- outgoing
+		middleware.mu.Unlock()
 		time.Sleep(5 * time.Second)
```

Comment:
Related to above, since this sends `MiddlewareInfo` every 5 seconds wouldn't it block the channel until it's read out of so you coulnd't send any other message types?